# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [9]:
import pandas as pd

### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [10]:
series1 = pd.Series(4, range(100))
series2 = pd.Series(2, range(100))
series3 = pd.Series(29845, range(100))

### Step 3. Let's create a DataFrame by joinning the Series by column

In [12]:
frame = {'Random1':series1, 'Random2':series2, 'Random3':series3}
dF = pd.DataFrame(frame)
print(dF)

    Random1  Random2  Random3
0         4        2    29845
1         4        2    29845
2         4        2    29845
3         4        2    29845
4         4        2    29845
..      ...      ...      ...
95        4        2    29845
96        4        2    29845
97        4        2    29845
98        4        2    29845
99        4        2    29845

[100 rows x 3 columns]


### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [15]:
dF.columns = ['bedrs', 'bathrs','price_sqr_meter']
print(dF)

    bedrs  bathrs  price_sqr_meter
0       4       2            29845
1       4       2            29845
2       4       2            29845
3       4       2            29845
4       4       2            29845
..    ...     ...              ...
95      4       2            29845
96      4       2            29845
97      4       2            29845
98      4       2            29845
99      4       2            29845

[100 rows x 3 columns]


### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [16]:
dF.stack().reset_index()

level_0          level_1      0
0          0            bedrs      4
1          0           bathrs      2
2          0  price_sqr_meter  29845
3          1            bedrs      4
4          1           bathrs      2
..       ...              ...    ...
295       98           bathrs      2
296       98  price_sqr_meter  29845
297       99            bedrs      4
298       99           bathrs      2
299       99  price_sqr_meter  29845

[300 rows x 3 columns]

### Step 6. Oops, it seems it is going only until index 99. Is it true?

In [ ]:
#No

### Step 7. Reindex the DataFrame so it goes from 0 to 299

In [ ]:
#Review step 5